cmd/terramate/cli/generate_test.go

```diff
@@ -622,6 +622,133 @@ stack {
 				},
 			},
 		},
+		{
+			name:   "multiple stacks with config on parent dir using globals from root",
+			layout: []string{"s:stacks/stack-1", "s:stacks/stack-2"},
+			configs: []backendconfig{
+				{
+					relpath: ".",
+					config: `
+globals {
+  bucket = "project-wide-bucket"
+}`,
+				},
+				{
+					relpath: "stacks",
+					config: `terramate {
+  backend "gcs" {
+    bucket = global.bucket
+    prefix = terramate.path
+  }
+}`,
+				},
+			},
+			want: want{
+				stacks: []stackcode{
+					{
+						relpath: "stacks/stack-1",
+						code: `terraform {
+  backend "gcs" {
+    bucket = "project-wide-bucket"
+    prefix = "/stacks/stack-1"
+  }
+}
+`,
+					},
+					{
+						relpath: "stacks/stack-2",
+						code: `terraform {
+  backend "gcs" {
+    bucket = "project-wide-bucket"
+    prefix = "/stacks/stack-2"
+  }
+}
+`,
+					},
+				},
+			},
+		},
+		{
+			name:   "stack with global on parent dir using config from root",
+			layout: []string{"s:stacks/stack"},
+			configs: []backendconfig{
+				{
+					relpath: ".",
+					config: `terramate {
+  backend "gcs" {
+    bucket = global.bucket
+    prefix = terramate.path
+  }
+}`,
+				},
+				{
+					relpath: "stacks",
+					config: `
+globals {
+  bucket = "project-wide-bucket"
+}`,
+				},
+			},
+			want: want{
+				stacks: []stackcode{
+					{
+						relpath: "stacks/stack",
+						code: `terraform {
+  backend "gcs" {
+    bucket = "project-wide-bucket"
+    prefix = "/stacks/stack"
+  }
+}
+`,
+					},
+				},
+			},
+		},
+		{
+			name:   "reference to undefined global fails",
+			layout: []string{"s:stack"},
+			configs: []backendconfig{
+				{
+					relpath: ".",
+					config: `terramate {
+  backend "gcs" {
+    bucket = global.bucket
+  }
+}`,
+				},
+			},
+			want: want{
+				res: runResult{
+					Error:        terramate.ErrGenerateBackendConfig,
+					IgnoreStdout: true,
+				},
+			},
+		},
+		{
+			name:   "invalid global definition fails",
+			layout: []string{"s:stack"},
+			configs: []backendconfig{
+				{
+					relpath: ".",
+					config: `terramate {
+  backend "gcs" {
+    bucket = "all good"
+  }
+}
+
+globals {
+  undefined_reference = global.undefined
+}
+`,
+				},
+			},
+			want: want{
+				res: runResult{
+					Error:        terramate.ErrGenerateLoadingGlobals,
+					IgnoreStdout: true,
+				},
+			},
+		},
 	}
```

Comment:
I know it's already covered in globals_test.go but maybe having the overriding/shadowing test here as well would be good, as this is an important behavior for the generate+globals.

Reply:
makes sense, also wanted to improve the tests overall x_x (bunch of duplication on HCL raw strings). WDYT on doing in a future PR ?

Reply:
Yep, makes sense